strax/run_selection.py

```diff
@@ -99,15 +99,16 @@ def scan_runs(self: strax.Context,
         _temp_docs = []
         for doc in sf._scan_runs(store_fields=store_fields):
             # If there is no number, make one from the name
+            _is_superrun = doc['name'].startswith('_')
             if 'number' not in doc:
                 if 'name' not in doc:
                     raise ValueError(f"Invalid run doc {doc}, contains "
                                      f"neither name nor number.")
-                _is_superrun = doc['name'].startswith('_')
                 if not _is_superrun:
-                    doc['number'] = int(doc['name'])
                     # If there is no name, make one from the number
-                    doc.setdefault('name', f"{doc['number']:06d}")
+                    doc['number'] = int(doc['name'])
+            if not _is_superrun:
```

Comment:
This should not even be strictly necessary right because a super-run always has a name

Reply:
> doc.setdefault('name', f"{doc['number']:06d}")

Was not touched by me this was there before.